website/project/model.py

```diff
@@ -1215,6 +1226,49 @@ def nodes_primary(self):
         ]
 
     @property
+    def depth(self):
+        return len(self.parents)
+
+    def next_descendants(self, auth, condition=lambda auth, node: True):
+        """
+        Recursively find the first set of descedants under a given node that meet a given condition
+
+        returns a list of [(node, [children]), ...]
+        """
+        ret = []
+        for node in self.nodes:
+            if condition(auth, node):
+                # base case
+                ret.append((node, []))
+            else:
+                ret.append((node, node.next_descendants(auth, condition)))
+        ret = [item for item in ret if item[1] or condition(auth, item[0])]  # prune empty branches
+        return ret
+
+    def get_descendants_recursive(self, include=lambda n: True):
+        for node in self.nodes:
+            yield node
+            for descedant in node.get_descendants_recursive(include):
+                yield descedant
+
+    def get_descendants_iterative(self, include=lambda n: True):
+        ret = []
+        stack = list(self.nodes)
+        while stack:
+            node = stack.pop()
+            if include(node):
+                ret.append(node)
+            stack = list(node.nodes) + stack
+        return ret
+
+    def get_aggregate_logs_set(self, auth):
```

Comment:
A bit of a misnomer. This returns a query set, not a set. How about just `get_aggregate_logs`?